c7n/resources/kafka.py

```diff
@@ -50,9 +57,37 @@ class KafkaSGFilter(SecurityGroupFilter):
 
 
 @Kafka.filter_registry.register('subnet')
-class KafkaSubnetFilter(SubnetFilter):
+class KafkaCompoundSubnetFilter(SubnetFilter):
 
-    RelatedIdsExpression = "BrokerNodeGroupInfo.ClientSubnets[]"
+    RelatedIdsExpression = "compound"
+
+    def process(self, resources, event=None):
+        # kafka v2 has both serverless and provisioned resources which have two different
+        # locations for their subnet info
+
+        class ProvisionedSubnetFilter(SubnetFilter):
+            RelatedIdsExpression = "Provisioned.BrokerNodeGroupInfo.ClientSubnets[]"
+
+        class ServerlessSubnetFilter(SubnetFilter):
+            RelatedIdsExpression = "Serverless.VpcConfigs[].SubnetIds"
+
+        p = []
+        s = []
+
+        for r in resources:
+            if r['ClusterType'] == 'PROVISIONED':
+                p.append(r)
+            if r['ClusterType'] == 'SERVERLESS':
+                s.append(r)
```

Comment:
can we get a server less cluster in the testing mix

Reply:
done